src/waitForTrigger.m

```diff
@@ -43,7 +43,7 @@ function waitForTrigger(varargin)
 
             keyCode = []; %#ok<NASGU>
 
-            [~, keyCode] = KbPressWait(deviceNumber);
+            [~, ~, keyCode] = KbCheck(deviceNumber);
```

Comment:
Hello @marcobarilari, I know you close this issue/merged this PR. But I want to ask two things.
1. I'm using this function to get timestamp for my onsets by setting this line: 
`[sec, keyCode] = KbPressWait(deviceNumber);`
and have the `sec` as output of this function. 
`sec = waitForTrigger(varargin)`

Then I call in my main exp script: 
`cfg.experimentStart = waitForTrigger(cfg)
`
Question here is do you see any logic mistake for storing onsets with this way of using the function?

2. When I update my submodule with this PR, how can I "cherry pick" ? :D @Remi-Gau

Reply:
Hello!

(1) I don't think that there is any "logic mistake" in using such secondary measures unless they are not in the right place in a script. (primary one would be like `startTime = PsychPortAudio(‘Start’, ...`). It depends what is your purpose.  

Though, I understand now why updating the submodels breaks your scripts :) and maybe here is the annoying thing that might tell you to not change function this way.

If I can suggest, you have the original `waitForTrigger` that, if I understand correctly, will pace your stimulation  and right after that you can use 

```matlab
    
     waitForTrigger(cfg)

    onset = GetSecs;

    *stimulation*

    estStopTime = GetSecs;

    duration = estStopTime - onset;

```

Reply:
> Though, I understand now why updating the submodels breaks your scripts :) and maybe here is the annoying thing that might tell you to not change function this way.

look who is being mean now.. I only changed this function - only today. all the other submodule issues are due to submodule itself 👯 

I'm using like this and it makes puurfect onsets (e.g. it's precisely right after my onsetDelay the stim starts) - without annoying decimal .00345 sort of things:

```
    % wait for trigger from fMRI
    cfg.experimentStart = waitForTrigger(cfg);
    ...
    % wait for baseline delays and then start the audio
    onset = PsychPortAudio('Start', cfg.audio.pahandle, [], ...
        cfg.experimentStart + cfg.timing.onsetDelay,1);
```
See my point? if I use GetSecs, then everything will be slightly weirdly numbered... 

Also @marcobarilari I see that you do not use any buffer for your audio exp, so no audio cracks? That's good but I wonder you are buffering somewhere in your audio-mot localiser?

Reply:
> I'm using like this and it makes puurfect onsets

it is actually a fair point.....

> I see that you do not use any buffer for your audio exp

we fill the buffer here

subfun/doAuditoryMotion.m line 46

```matlab
% Start the sound presentation
PsychPortAudio('FillBuffer', cfg.audio.pahandle, sound);
PsychPortAudio('Start', cfg.audio.pahandle);
onset = GetSecs;
```

Reply:
yup i saw that one and i was using that as well `FillBuffer` and `Start`... then in every first run, in the very first block there were audio cracks, so I'm using `Start` with option (see below). Then I remembered during my aud-mot exp, i was also inserting such option:
```
PsychPortAudio('Start', cfg.audio.pahandle, [], ...
        cfg.experimentStart + cfg.timing.onsetDelay,1)
```
Just to warn you to pay attention to audio  ｡◕‿◕｡

btw, if you see my point,...does it mean you would consider changing the `waitForTrigger` for such optional use? ♥‿♥

Reply:
Big dilemma :) can we pause this discussion for a couple of days? Maybe let me move this conversation in an issue and let's come back to it later.